module/assets/helper.js

```diff
@@ -0,0 +1,353 @@
+                  let insertProgramRolesQuery = {
+                    userId: reqData.toUserProfile.userId,
+                    programRoles: [],
+                  };
+                  let insertProgram = await userExtensionsHelper.insertOne(
+                    insertProgramRolesQuery
+                  );
+
+                  if (insertProgram) {
+                    let updateProgram;
+                    if (
+                      reqData.toUserProfile.role === "PROGRAM_MANAGER" ||
+                      reqData.toUserProfile.role === "PROGRAM_DESIGNER"
+                    ) {
+                      await userExtensionsHelper.updateOne(updateQuery);
+                    }
+                    if (updateProgram) {
+                      let deleteProgram = await userExtensionsHelper.updateOne(
+                        fromUpdateQuery
+                      );
+                      // resData=updateProgram;
+                      // return resolve(updateProgram);
+                    }
+                  }
+                } else {
+                  let updateProgram;
+                  if (
+                    reqData.toUserProfile.role === "PROGRAM_MANAGER" ||
+                    reqData.toUserProfile.role === "PROGRAM_DESIGNER"
+                  ) {
+                    await userExtensionsHelper.updateOne(updateQuery);
+                  }
+                  if (updateProgram) {
+                    let deleteProgram = await userExtensionsHelper.updateOne(
+                      fromUpdateQuery
+                    );
+                    // resData=updateProgram;
+
+                    // return resolve(updateProgram);
+                  }
+                }
+              } else {
+                let findProgramManagerQuery = {
+                  code: { $in: reqData.toUserProfile.role },
+                };
+                let findProgramManagerId;
+                findProgramManagerId = await userRolesHelper.findOne(
+                  findProgramManagerQuery
+                );
+
+                const roleToUpdate =
+                  newCollectionForUserExtension.platformRoles.find(
+                    (role) =>
+                      role.code ===
+                      reqData.toUserProfile.role.includes(role.code)
+                  );
+                if (roleToUpdate) {
+                  roleToUpdate.roleId = findProgramManagerId._id;
+                  newCollectionForUserExtension.platformRoles =
+                    newCollectionForUserExtension.platformRoles.filter(
+                      (role) =>
+                        role.code !==
+                        reqData.toUserProfile.role.includes(role.code)
+                    );
+                }
+                let createUserExtensions = await userExtensionsHelper.createOne(
+                  newCollectionForUserExtension
+                );
+                if (createUserExtensions) {
+                  ownershipTransfer(ownershipTransferEvent);
```

Comment:
this will create recursion dont you think @praveenKDass

Reply:
resolved